File: examples/python/nlgpio16.py
#!/usr/bin/python
# Author: Jon Trulson <jtrulson@ics.com>
# Copyright (c) 2015 Intel Corporation.
#
# Permission is hereby granted, free of charge, to any person obtaining
# a copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, sublicense, and/or sell copies of the Software, and to
# permit persons to whom the Software is furnished to do so, subject to
# the following conditions:
#
# The above copyright notice and this permission notice shall be
# included in all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

from __future__ import print_function
import sys, signal, atexit
from upm import pyupm_nlgpio16 as sensorObj

def main():
    # Instantiate a NLGPIO16 Module on the default UART (/dev/ttyACM0)
    sensor = sensorObj.NLGPIO16()

    ## Exit handlers ##
    # This stops python from printing a stacktrace when you hit control-C
    def SIGINTHandler(signum, frame):
        raise SystemExit

    # This function lets you run code on exit
    def exitHandler():
        print("Exiting")
        sys.exit(0)

    # Register exit handlers
    atexit.register(exitHandler)
    signal.signal(signal.SIGINT, SIGINTHandler)

    # get the Version
    print("Device Version:", sensor.getVersion())
    # read the gpio at pin 3
    print("GPIO 3 Value:", sensor.gpioRead(3))
    # read the analog voltage at pin 5
    print("Analog 5 Voltage:", sensor.analogReadVolts(5))
    # set the gpio at pin 14 to HIGH
    sensor.gpioSet(14)

if __name__ == '__main__':
    main()

File: examples/javascript/nlgpio16.js
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */


var sensorObj = require('jsupm_nlgpio16');

// Instantiate a NLGPIO16 Module on the default UART (/dev/ttyACM0)
var sensor = new sensorObj.NLGPIO16();

// get the Version
console.log("Device Version:", sensor.getVersion());
// read the gpio at pin 3
console.log("GPIO 3 Value:", sensor.gpioRead(3));
// read the analog voltage at pin 5
console.log("Analog 5 Voltage:", sensor.analogReadVolts(5));
// set the gpio at pin 14 to HIGH
sensor.gpioSet(14);

/************** Exit code **************/
process.on('SIGINT', function()
{
    sensor = null;
    sensorObj.cleanUp();
    sensorObj = null;
    console.log("Exiting...");
    process.exit(0);
});
